frontend/public/components/build-pipeline.tsx

```diff
@@ -0,0 +1,174 @@
+    running: 'fa-refresh fa-spin',
+    complete: 'fa-check-circle',
+    failed: 'fa-times-circle'
+  })[statusClass];
+  const hide = icon ? 'hide' : '';
+  return <React.Fragment>
+    <span className={`build-pipeline__status-icon build-pipeline__status-icon--${statusClass}`}>
+      <span aria-hidden="true">
+        <span className={`fa ${icon} fa-fw`}></span>
+      </span>
+    </span>
+
+    <span className={`build-pipeline__status-icon ${hide}`}>
+      <span className="fa fa-refresh fa-spin" aria-hidden="true"></span>
+    </span>
+  </React.Fragment>;
+};
+
+const BuildLogLink: React.SFC<BuildLogLinkProps> = ({obj}) => {
+  const link = getJenkinsLogURL(obj);
+  return link ? <div className="build-pipeline__link">
+    <a href={link} className="build-pipeline__log-link" target="_blank" rel="noopener noreferrer">View Log</a>
+  </div> : null;
+};
+
+const StagesNotStarted: React.SFC<StagesNotStartedProps> = ({stages}) => {
+  return !stages.length ? <div className="build-pipeline__stage build-pipeline__stage--none">No stages have started.
+  </div> : null;
+};
+
+
+const BuildSummaryTimestamp: React.SFC<BuildSummaryTimestampProps> = ({timestamp}) => <span className="build-pipeline__timestamp text-muted">
+  { fromNow(timestamp) }
+</span>;
+
+const BuildPipelineSummary: React.SFC<BuildPipelineSummaryProps> = ({obj}) => {
+  const buildNumber = getBuildNumber(obj);
+  return <div className="build-pipeline__summary">
+    <div className="build-pipeline__phase">
+      <BuildSummaryStatusIcon status={obj.status.phase} /> Build {buildNumber}
+    </div>
+    <BuildSummaryTimestamp timestamp={obj.metadata.creationTimestamp} />
+    <BuildLogLink obj={obj} />
+  </div>;
+};
+
+const BuildAnimation:React.SFC<BuildAnimationProps> = ({status}) => <div className={`build-pipeline__status-bar build-pipeline__status-bar--${_.kebabCase(status)}`}>
+  <div className="build-pipeline__animation-line"></div>
+  <div className="build-pipeline__animation-circle">
+    <div className="build-pipeline__circle-clip1"></div>
+    <div className="build-pipeline__circle-clip2"></div>
+    <div className="build-pipeline__circle-inner">
+      <div className="build-pipeline__circle-inner-fill"></div>
+    </div>
+  </div>
+</div>;
+
+const JenkinsInputUrl: React.SFC<JenkinsInputUrlProps> = ({obj, stage}) => {
+  const pending = stage && (stage.status === 'PAUSED_PENDING_INPUT');
+  const buildUrl = getJenkinsBuildURL(obj);
+  if(pending) {
+    return <div className="build-pipeline__stage-actions text-muted">
+      <a href={buildUrl} target="_blank" rel="noopener noreferrer">Input Required</a>
+    </div>;
+  }
+  return null;
+};
+
+const BuildStageTimestamp:React.SFC<BuildStageTimestampProps> = ({timestamp}) => <div className="build-pipeline__stage-time text-muted">
+  { fromNow(timestamp) }
+</div>;
+
+const BuildStageName: React.SFC<BuildStageNameProps> = ({name}) => {
+  return <div title={name} className="build-pipeline__stage-name">
```

Comment:
`title` is because long names get elided?

Reply:
Think so:

```css
.build-pipeline__stage-name {
  @include text-overflow();
  margin-bottom: $pipeline-padding + 3px;
}
```

Reply:
that's correct